asf_public_discourse_home_decarbonisation/pipeline/faqs_identification/extract_questions.py

```diff
@@ -0,0 +1,445 @@
+
+    Returns:
+        sentences_without_inclusion (List[str]):
+            - This list contains sentences extracted from the "I don't know" phrases, excluding the inclusion phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the inclusion phrases.
+        - If an inclusion phrase is not found in a sentence, that sentence will not be included in the output list.
+    """
+    sentences_without_inclusion = []
+    for sentence in idk_phrases:
+        for phrase in inclusion_phrases:
+            if phrase.lower() in sentence.lower():
+                start_index = sentence.lower().index(phrase.lower()) + len(phrase)
+                extracted_text = sentence[start_index:].strip()
+                sentences_without_inclusion.append(extracted_text)
+                break
+    return sentences_without_inclusion
+
+
+def logger_statistics(category_dataframe):
+    """
+    Calculates and logs basic statistics about the category-specific data.
+
+    Args:
+        category_dataframe (DataFrame): The category-specific data.
+
+    Note:
+        - The function calculates the total number of "idk" phrases, questions, and sentences in the dataframe.
+        - The statistics are logged using the `logging.info()` method.
+
+    Example:
+        logger_statistics(category_dataframe)
+
+    """
+    total_number_of_idk_phrases = category_dataframe["idk_phrases"].apply(len).sum()
+    total_number_of_sentences = category_dataframe["text"].apply(len).sum()
+    total_questions = category_dataframe["questions"].apply(len).sum()
+
+    # Print some basic statistics
+    logging.info(f"Total number of idk phrases: {total_number_of_idk_phrases}")
+    logging.info(f"Total number of questions: {total_questions}")
+    logging.info(f"Total number of sentences: {total_number_of_sentences}")
+
+
+if __name__ == "__main__":
+    # Read the CSV file
+    args = create_argparser()
+    category = args.category
+    collection_date_time = args.collection_date_time
+    num_of_words = args.num_of_words
+    post_type = args.post_type
+    # uses different getter function depending on the forum
+    category_dataframe = get_category_data(args.forum, category, collection_date_time)
+    FORUM_FAQ_PATH = os.path.join(
+        PROJECT_DIR, f"outputs/data/extracted_questions/{args.forum}/forum_{category}/"
+    )
+    # Ensure the output directory exists
+    os.makedirs(FORUM_FAQ_PATH, exist_ok=True)
+    question_words = [
+        "what",
+        "how",
+        "where",
+        "when",
+        "who",
+        "why",
+        "which",
+        "really",
+        "any",
+        "what's",
+        "how's",
+        "really",
+        "thoughts",
+        "is",
+    ]
+    # List of phrases to include
+    inclusion_phrases = [
+        "don’t know ",
+        "do not know ",
+        "do not know how to ",
+    ]
```

Comment:
When we search for "do not know" we will also get "do not know how to", so in theory we don't need both. But it's good to keep both because when we extract what comes after the expressions (our X), we want the "how to" to not appear. In order for it to work, we need to change the order by which the inclusion_phrases are defined, so that when we apply the function `extract_X_from_idk`,  the "do not know how to" comes up before "do not know". Let me know if that does not makes sense. I've also added "don't know how to" to the list.

```suggestion
    inclusion_phrases = [
        "don't know how to ",
        "don’t know ",
        "do not know how to ",
        "do not know ",
    ]
```

Reply:
Yep it makes sense. Might be simpler to just have "don't know" and "do not know" because I think this would make things a little bit simpler when interpreting the data? Might be just me though.

Reply:
yep, sounds good. i like that